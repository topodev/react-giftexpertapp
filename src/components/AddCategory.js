import React, { useState } from 'react';
import Proptypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        
        setInputValue(e.target.value)
    }

    const handlesubmit = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length>2){

            setCategories( cats=>[ inputValue, ...cats ]);
            setInputValue('');
        }

        

    }

    return (
       
            <form onSubmit={handlesubmit}>
            <input
            type="text"
            value={inputValue}
            onChange={ handleInputChange}         
            />
            
            </form>
            
        
    )
}

AddCategory.prototype={
    setCategories: Proptypes.func.isRequired
}
