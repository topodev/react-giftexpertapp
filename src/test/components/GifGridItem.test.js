import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () =>{

    test('should show the component correctly ', () => {

        const wrapper= shallow( <GifGridItem/>)
        expect( wrapper ).toMatchSnapshot();
        
    })
    
})