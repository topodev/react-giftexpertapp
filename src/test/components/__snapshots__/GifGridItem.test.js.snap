// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> should show the component correctly  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img />
  <p />
</div>
`;
